### PR TITLE
[Validator] Update DateTimeValidator to validate with format only

### PR DESCRIPTION
**Pull request by lepiaf**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #11925
| License       | MIT
| Doc PR        | symfony/symfony-docs#7583

- [x] update datetime validator
- [x] update documentation for datetime validator
- [x] add deprecation on DateValidator, TimeValidator
- [x] update UPGRADE file

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Thanks for this contribution!

I've read the related discussion about I'm still not convinced about the `Timestamp` name. It's true that technically speaking, any date + time is a "timestamp" (see https://en.wikipedia.org/wiki/Timestamp) but I'm sure that when a developers see a "TimestampValidator", they are thinking about UNIX timestamps (e.g. `1256953732`).

Don't you think that "DateTime" is a better name? A "DateTime" can be a "date", a "time", a "date + time", a "UNIX timestamp", etc.

File: src/Symfony/Component/Validator/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 CHANGELOG
 =========
 
+3.4.0
```
Comment (HeahDude) [inline src/Symfony/Component/Validator/CHANGELOG.md]: It's still 3.3 :), same for all deprecation notices.

**Comment by lepiaf** [comment]
I can rework DateTime validator to be more flexible and deprecate Date and Time validators only.

File: src/Symfony/Component/Validator/Constraints/Time.php [hunk 2]
```diff
@@ -18,6 +18,8 @@
  * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
+ *
+ * @deprecated Since version 3.3, to be removed in 4.0. Use Symfony\Component\Validator\Constraints\DateTimeValidator instead.
```
Comment (stof) [inline src/Symfony/Component/Validator/Constraints/Time.php]: you also need to add the E_USER_DEPRECATED on top of the file

File: src/Symfony/Component/Validator/Constraints/DateTime.php [hunk 3]
```diff
@@ -31,6 +39,22 @@ class DateTime extends Constraint
...
+    {
+        return 'format';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequiredOptions()
+    {
+        return array('format');
```
Comment (stof) [inline src/Symfony/Component/Validator/Constraints/DateTime.php]: making it required is a BC break. It should keep the existing default

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ CHANGELOG
 
  * added `AddValidatorInitializersPass`
  * added `AddConstraintValidatorsPass`
+ * deprecate `DateValidator` in favor of `DateTimeValidator`
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/CHANGELOG.md]: deprecated

File: src/Symfony/Component/Validator/CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,8 @@ CHANGELOG
...
+ * deprecate `TimeValidator` in favor of `DateTimeValidator`
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/CHANGELOG.md]: same here

**Comment by lepiaf** [comment]
any comment about this pr ? should I change to branch 3.4 ?

**Review by ro0NL — COMMENTED** [review]
Should be 3.4 yes :)

What aboutd deprecating `\DateTime` instances as well? It's weird to use a datetime validator for an already valid datetime object.

Perhaps add constants for pre-defined formats, that be nice :) i.e. FORMAT_DATE/TIME/TIME_WITHOUT_SECONDS

**Comment by lepiaf** [comment]
As I don't have time to work on this PR, I close it. Feel free to continue it :)

**Pull request closed without merging**